syntax = "proto3";

package kvstore;

service KeyValueStore {
    rpc Get(GetRequest) returns (GetResponse) {}
    rpc Put(PutRequest) returns (PutResponse) {}
}

// You'll likely need to define more specific return codes than these!
enum ReturnCode {
    SUCCESS = 0;
    FAILURE = 1;
}

message GetRequest {
    string key = 1;
    string flag = 2;
    int32 id = 3;
}

message GetResponse {
    string value = 1;
    ReturnCode ret = 2;
}

message PutRequest {
    string key = 1;
    string value = 2;
    string flag = 3;
    int32 id = 4; 
}

message PutResponse {
    ReturnCode ret = 1;
}